doc/devdocs/ast.rst

```diff
@@ -183,12 +183,65 @@ These symbols appear in the ``head`` field of ``Expr``\s in lowered form.
 
     ``:pop_loc``: returns to the source location before the matching ``:push_loc``.
 
-LambdaInfo
+
+Method
+~~~~~~
+
+A unique'd container describing the shared metadata for a single (unspecialized) method.
+
+``name``, ``module``, ``file``, ``line``, ``sig`` - Metadata to uniquely identify the method
+    for the computer and the human
+
+``ambig`` - Cache of other methods that may be ambiguous with this one
+
+``specializations`` - Cache of all MethodInstance ever created for this Method,
+    used to ensure uniqueness. Uniqueness is required for efficiency,
+    especially for incremental precompile and tracking of method invalidation.
+
+``source`` - The original source code (compressed),
+
+``roots`` - Pointers to non-AST things required by compression of the AST or native code from codegen.
```

Comment:
We allow embedding of arbitrary objects in the AST. It's impossible to give a complete list.